consensus/safety-rules/src/safety_rules.rs

```diff
@@ -89,17 +97,46 @@ impl SafetyRules {
         }
     }
 
+    /// This reconciles the key pair of a validator signer with a given validator set
+    /// during epoch changes.
+    /// Current impl of consensus does not expect key reconciliation to panic or fail.
+    /// Instead, we signal debug messages indicating following error causes:
+    ///     1. Validator not in the set
+    ///     2. Validator in the set, but no matching key found in storage
+    fn reconcile_key(&mut self, epoch_state: &EpochState) -> Result<(), Error> {
+        if let Some(expected_key) = epoch_state.verifier.get_public_key(&self.author) {
+            let curr_key = self.signer().ok().map(|s| s.public_key());
+            if curr_key != Some(expected_key.clone()) {
+                debug!(
+                    "Reconciled pub key for signer {} [{:#?} -> {:#?}]",
```

Comment:
don't the keys support display?, I think you can just do {} -> {}

Reply:
curr_key is of Option; fixed in latest commit